core/src/main/java/org/apache/accumulo/core/fate/zookeeper/FateLock.java

```diff
@@ -78,11 +78,11 @@ public static class FateLockNode implements Comparable<FateLockNode> {
     public final long sequence;
     public final String lockData;
 
-    private FateLockNode(String nodeName) {
+    FateLockNode(String nodeName) {
       int len = nodeName.length();
       Preconditions.checkArgument(nodeName.startsWith(PREFIX) && nodeName.charAt(len - 11) == '#',
           "Illegal node name %s", nodeName);
```

Comment:
In the case of an overflow, the character at (len - 11) be a minus sign and cause this to fail.

Reply:
It depends on the number of digits in the negative number.  If its a 10 digit negative number then `%010d` will produce a 11 char string.  If its 9 digits or less then `%010d` will produce a 10 char string including the negative.  For the 9 digit or less case the `#` check will not catch negative, but the later `Long.parseUnsignedLong` call will catch it.

Reply:
For the case Long.parseUnsignedLong catches the problem it throws a more specific exception,  I updated the test to check for that.